sycl/test/usm/pointer_query.cpp

```diff
@@ -92,5 +92,27 @@ int main() {
   }
   free(array, ctxt);
 
+  // Test invalid ptrs
+  Kind = get_pointer_type(nullptr, ctxt);
+  if (Kind != usm::alloc::unknown) {
+    return 11;
+  }
+
+  // next checks only valid for non-host contexts
```

Comment:
Is this because the host context will return `usm::alloc::host` instead of `unknown` for the system `malloc`?

If so, this wasn't obvious to me on first reading.  I'd recommend adding an explanation to the comment or testing the expected return type for host contexts in an else.

Reply:
Yes.